src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy

```diff
@@ -0,0 +1,76 @@
+package org.gradle.android.workarounds
+
+import org.gradle.android.AndroidIssue
+import org.gradle.api.Task
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.process.CommandLineArgumentProvider
+
+/**
+ * Changes annotation processor arguments to set the room.schemaLocation via a CommandLineArgsProcessor
+ * which accurately sets it as an output of the task.  Note that it is common for people to set the value
+ * using the defaultConfig in the android extension, which causes every compile task to use the same
+ * output directory, causing overlapping outputs and breaking cacheability again.  To avoid this, we
+ * change the location for each task to treat the specified directory as a "base" directory, but derive
+ * the task-specific directory off of the task path, ensuring uniqueness across compile tasks.
+ */
+@AndroidIssue(introducedIn = "3.5.0", fixedIn = [], link = "https://issuetracker.google.com/issues/132245929")
+class RoomSchemaLocationWorkaround implements Workaround {
```

Comment:
This looks great. I was curious, while we're fixing this cache issue, does it also make sense to mark this schemas directory as an input to the test and androidTest tasks? Otherwise changes to the schema might not re-run the test if the tests are cached.

Reply:
That's a good suggestion.  I'll check it out and possibly implement that in a follow up change.  For now, I want to get an initial version of this out and see how well the existing fixes work.